File: src/theme/designSystem.ts
import { FontScale, ThemeMode } from '../types';

// Prominent Slate Light Gray + Electric Cyan Palette
export const COLORS = {
  // Electric Cyan Core Palette
  cyanPrimary: '#0284C7',
  cyanLight: '#38BDF8',
  cyanDark: '#0369A1',
  cyanGlow: 'rgba(2, 132, 199, 0.25)',

  // Unmistakable Prominent Slate Gray Backgrounds & Cards
  lightGrayBg: '#64748B',      // Distinct Slate Gray Background
  lightGrayCard: '#94A3B8',    // Prominent Slate Gray Card
  lightGrayBorder: '#475569',  // Darker Slate Border
  pureWhite: '#FFFFFF',
  
  // High-Contrast Typography
  darkText: '#0F172A',
  mutedText: '#334155',
  darkGray: '#1E293B',

  // Status & Safety
  uberSafetyRed: '#E11D48',
  uberSafetyGreen: '#0284C7',
  uberWarningAmber: '#F59E0B',

  // Mic Button Accents (Cyan Buttons)
  micIdleBg: '#0284C7',
  micActiveBg: '#38BDF8',
  micActiveBorder: '#0284C7',
  micActiveGlow: 'rgba(56, 189, 248, 0.45)',

  // Legacy Aliases
  primaryBlue: '#0284C7',
  softGreen: '#38BDF8',
  safetyCoral: '#E11D48',
  highContrastYellow: '#FACC15',
  mutedGray: '#475569',
  uberBlack: '#0F172A',
  uberWhite: '#FFFFFF',
  uberDarkCard: '#94A3B8',
  uberDarkHeader: '#64748B',
};

export const getThemeColors = (mode: ThemeMode) => {
  switch (mode) {
    case 'highContrastDark':
      return {
        bgCard: '#000000',
        bgHeader: '#000000',
        textPrimary: '#FFFFFF',
        textSecondary: '#FACC15',
        accent: '#38BDF8',
        border: 'transparent',
        buttonPrimary: '#0284C7',
        buttonText: '#FFFFFF',
        activeTab: '#38BDF8',
        hazardBg: '#FF0000',
        hazardText: '#FFFFFF',
      };
    case 'highContrastAmber':
      return {
        bgCard: '#181100',
        bgHeader: '#0A0700',
        textPrimary: '#FFD700',
        textSecondary: '#FFA500',
        accent: '#FFD700',
        border: 'transparent',
        buttonPrimary: '#FFD700',
        buttonText: '#000000',
        activeTab: '#FFD700',
        hazardBg: '#FF4500',
        hazardText: '#FFFFFF',
      };
    case 'standard':
    default:
      return {
        bgCard: '#64748B',
        bgHeader: '#64748B',
        textPrimary: '#0F172A',
        textSecondary: '#334155',
        accent: '#0284C7',
        border: '#475569',
        buttonPrimary: '#0284C7',
        buttonText: '#FFFFFF',
        activeTab: '#0284C7',
        hazardBg: '#E11D48',
        hazardText: '#FFFFFF',
      };
  }
};

export const getFontSizes = (scale: FontScale) => {
  const multiplier = scale === 'extraLarge' ? 1.3 : scale === 'large' ? 1.15 : 1.0;
  return {
    xs: Math.round(13 * multiplier),
    sm: Math.round(15 * multiplier),
    base: Math.round(17 * multiplier),
    lg: Math.round(20 * multiplier),
    xl: Math.round(24 * multiplier),
    xxl: Math.round(30 * multiplier),
    hero: Math.round(38 * multiplier),
  };
};

export const ACCESSIBILITY = {
  minTouchTargetSize: 52,
  largeTouchTargetSize: 64,
  fabMicSize: 68,
  borderRadiusCard: 22,
  borderRadiusButton: 16,
  borderRadiusChip: 999,
};

File: navidoor_flutter/lib/theme/design_system.dart
import 'package:flutter/material.dart';

enum ThemeModeOption {
  standard,
  highContrastDark,
  highContrastAmber,
}

enum FontScaleOption {
  normal,
  large,
  extraLarge,
}

class AppColors {
  // Electric Cyan Core Palette
  static const Color cyanPrimary = Color(0xFF0284C7);
  static const Color cyanLight = Color(0xFF38BDF8);
  static const Color cyanDark = Color(0xFF0369A1);
  static const Color cyanGlow = Color(0x400284C7);

  // Unmistakable Prominent Slate Gray Backgrounds & Cards
  static const Color lightGrayBg = Color(0xFF64748B);     // Distinct Slate Gray Background
  static const Color lightGrayCard = Color(0xFF94A3B8);   // Prominent Slate Gray Card
  static const Color lightGrayBorder = Color(0xFF475569); // Darker Slate Border
  static const Color pureWhite = Color(0xFFFFFFFF);
  
  // High-Contrast Typography
  static const Color darkText = Color(0xFF0F172A);
  static const Color mutedText = Color(0xFF334155);
  static const Color darkGray = Color(0xFF1E293B);

  // Status & Safety
  static const Color uberSafetyRed = Color(0xFFE11D48);
  static const Color uberSafetyGreen = Color(0xFF0284C7);
  static const Color uberWarningAmber = Color(0xFFF59E0B);

  // Mic Button Accents (Cyan Buttons)
  static const Color micIdleBg = Color(0xFF0284C7);
  static const Color micActiveBg = Color(0xFF38BDF8);
  static const Color micActiveBorder = Color(0xFF0284C7);
  static const Color micActiveGlow = Color(0x7338BDF8);

  // Aliases
  static const Color primaryBlue = Color(0xFF0284C7);
  static const Color softGreen = Color(0xFF38BDF8);
  static const Color safetyCoral = Color(0xFFE11D48);
  static const Color highContrastYellow = Color(0xFFFACC15);
  static const Color mutedGray = Color(0xFF475569);
  static const Color uberBlack = Color(0xFF0F172A);
  static const Color uberWhite = Color(0xFFFFFFFF);
  static const Color uberDarkCard = Color(0xFF94A3B8);
  static const Color uberDarkHeader = Color(0xFF64748B);
}

class ThemeColors {
  final Color bgCard;
  final Color bgHeader;
  final Color textPrimary;
  final Color textSecondary;
  final Color accent;
  final Color border;
  final Color buttonPrimary;
  final Color buttonText;
  final Color activeTab;
  final Color hazardBg;
  final Color hazardText;

  const ThemeColors({
    required this.bgCard,
    required this.bgHeader,
    required this.textPrimary,
    required this.textSecondary,
    required this.accent,
    required this.border,
    required this.buttonPrimary,
    required this.buttonText,
    required this.activeTab,
    required this.hazardBg,
    required this.hazardText,
  });

  static ThemeColors getThemeColors(ThemeModeOption mode) {
    switch (mode) {
      case ThemeModeOption.highContrastDark:
        return const ThemeColors(
          bgCard: Color(0xFF000000),
          bgHeader: Color(0xFF000000),
          textPrimary: Color(0xFFFFFFFF),
          textSecondary: Color(0xFFFACC15),
          accent: Color(0xFF38BDF8),
          border: Colors.transparent,
          buttonPrimary: Color(0xFF0284C7),
          buttonText: Color(0xFFFFFFFF),
          activeTab: Color(0xFF38BDF8),
          hazardBg: Color(0xFFFF0000),
          hazardText: Color(0xFFFFFFFF),
        );
      case ThemeModeOption.highContrastAmber:
        return const ThemeColors(
          bgCard: Color(0xFF181100),
          bgHeader: Color(0xFF0A0700),
          textPrimary: Color(0xFFFFD700),
          textSecondary: Color(0xFFFFA500),
          accent: Color(0xFFFFD700),
          border: Colors.transparent,
          buttonPrimary: Color(0xFFFFD700),
          buttonText: Color(0xFF000000),
          activeTab: Color(0xFFFFD700),
          hazardBg: Color(0xFFFF4500),
          hazardText: Color(0xFFFFFFFF),
        );
      case ThemeModeOption.standard:
        return const ThemeColors(
          bgCard: Color(0xFF64748B),
          bgHeader: Color(0xFF64748B),
          textPrimary: Color(0xFF0F172A),
          textSecondary: Color(0xFF334155),
          accent: Color(0xFF0284C7),
          border: Color(0xFF475569),
          buttonPrimary: Color(0xFF0284C7),
          buttonText: Color(0xFFFFFFFF),
          activeTab: Color(0xFF0284C7),
          hazardBg: Color(0xFFE11D48),
          hazardText: Color(0xFFFFFFFF),
        );
    }
  }
}

class FontSizes {
  final double xs;
  final double sm;
  final double base;
  final double lg;
  final double xl;
  final double xxl;

  const FontSizes({
    required this.xs,
    required this.sm,
    required this.base,
    required this.lg,
    required this.xl,
    required this.xxl,
  });

  static FontSizes getFontSizes(FontScaleOption scale) {
    final multiplier = scale == FontScaleOption.extraLarge
        ? 1.3
        : scale == FontScaleOption.large
            ? 1.15
            : 1.0;
    return FontSizes(
      xs: 13 * multiplier,
      sm: 15 * multiplier,
      base: 17 * multiplier,
      lg: 20 * multiplier,
      xl: 24 * multiplier,
      xxl: 28 * multiplier,
    );
  }
}

class ResponsiveHelper {
  static bool isMobile(BuildContext context) =>
      MediaQuery.of(context).size.width < 768;

  static bool isTablet(BuildContext context) =>
      MediaQuery.of(context).size.width >= 768 &&
      MediaQuery.of(context).size.width < 1024;

  static bool isDesktop(BuildContext context) =>
      MediaQuery.of(context).size.width >= 1024;

  static double screenWidth(BuildContext context) =>
      MediaQuery.of(context).size.width;

  static double screenHeight(BuildContext context) =>
      MediaQuery.of(context).size.height;
}
